### DISCUSSION TITLE
Limits to etcd cluster size

### DISCUSSION THREAD

**Discussion by arjunsingriuber** [etcd-io/etcd#18004] [Q&A] [answered]
Have there been any tests around how many hosts etcd can replicate to using RAFT?

**Comment by jmhbnz** [comment]
Hi @arjunsingriuber - Thanks for your question, please refer to our FAQ: https://etcd.io/docs/v3.5/faq/#what-is-maximum-cluster-size

> Theoretically, there is no hard limit. However, an etcd cluster probably should have no more than seven nodes. [Google Chubby lock service](http://static.googleusercontent.com/media/research.google.com/en//archive/chubby-osdi06.pdf), similar to etcd and widely deployed within Google for many years, suggests running five nodes. A 5-member etcd cluster can tolerate two member failures, which is enough in most cases. Although larger clusters provide better fault tolerance, the write performance suffers because data must be replicated across more machines.

**Reply by arjunsingriuber** [reply]
Is there a limit on the number of observers?

**Reply by jmhbnz** [reply]
To clarify are you referring to a cluster **learner** member? Or etcd **watch** functionality?

**Reply by arjunsingriuber** [reply]
Sorry for the confusion. I am talking about etcd watch functionality.

**Reply by jmhbnz** [reply, marked as answer]
Thanks for the clarification, etcd is well known for the efficiency of it's watch capabilities, refer https://etcd.io/docs/v3.5/benchmarks/etcd-3-watch-memory-benchmark noting that those documented benchmarks are now getting old.

This is something you can perform basic benchmarks in your own environment with https://github.com/etcd-io/etcd/tree/main/tools/benchmark. I would encourage you to perform validation in your environment to properly understand the performance characteristics in your environment.

**Reply by arjunsingriuber** [reply]
Thank you
